opm/output/data/Wells.hpp

```diff
@@ -99,8 +102,8 @@ namespace Opm {
             bool operator==(const Rates& rat2) const;
 
         private:
-            double& get_ref( opt );
-            const double& get_ref( opt ) const;
+            double& get_ref( opt, const std::string tracer_name = "" );
+            const double& get_ref( opt, const std::string tracer_name = "" ) const;
```

Comment:
I agree with @joakim-hove.  We should keep tracers out of the common path.  I'd prefer instead that we add dedicated functions to assign and retrieve tracer-related rates.  As another&mdash;minor&mdash;issue please don't take `std::string` parameters by `const` value.  If we're only meant to read from the string the parameter type should be
```C++
const std::string&
```
(reference to a constant string) or, if we're slightly more daring, a [`std::string_view`](https://en.cppreference.com/w/cpp/string/basic_string_view) from C++17 instead.

Reply:
Could 
```
set_tracer(...)
get_tracer(...)
get_ref_tracer(...)
```
work?
Where get/set/ref_tracer calls the usual get/set/ref code + does the tracer specific stuff. 
It will require some if(tracer) on the call site.

Reply:
I guess that could work. My feeling of the situation goes as:

1. The system for transferring data between the simulator and the output layer has grown large and unwieldy - not related to this PR.
2. The existing functions should not be exposed to tracer stuff. 
3. The existing implementation is unwieldy and a refactor is overdue - the tracer stuff can very well be implemented "differently" from what is already there - if that makes sense.